tests/subsys/tracing/tracing_api/src/main.c

```diff
@@ -230,6 +245,30 @@ void test_tracing_data_format(void)
 	tracing_format_raw_data(tracing_raw_data.data, tracing_raw_data.length);
 	k_sleep(K_MSEC(100));
 	zassert_true(raw_data_format_found == true, "Failded to check output from backend");
+
+	/* make tracing buffer full */
+	uint8_t *buf = data;
+
+	while (tracing_buffer_put_claim(&buf, 5)) {
+		;
+	}
+
```

Comment:
It is better to change the code like
do {
  ret = tracing_buffer_put_clain();
}while (ret)